File: EjerciciosPython/Ejercicio02/Coche.py
# -*- coding: utf-8 -*-
"""
Created on Tue Jan 10 19:31:06 2023

@author: Usuario
"""
from Vehiculo import *

class Coche(Vehiculo):
    
    def __init__(self, ide, marca, modelo, km, precio, airbag):
        super().__init__(ide, marca,modelo,km,precio)
        self.__airbag = airbag
    
    def getPrecio(self):
        if(self.__airbag == True):
            return super().getPrecio() + 200
        else:
            return super().getPrecio()

    def __str__(self):
        if(self.__airbag == True):    
            return super().__str__() + "\nSi tiene bolsas de aire"
        else:
            return super().__str__() + "\nNo tiene bolsas de aire "

File: Ejercicio06/Coche.cs
using System;

namespace Ejercicio06
{
    class Coche : Vehiculo
    {
        private bool _airbag;

        public Coche (string marca, string modelo, int km, int precio, bool airbag) : base(marca, modelo, km, precio)
        {
            _airbag = airbag;
        }

        //Getter - Setter de airbag
        public bool airbag
        {
            get
            {
                return _airbag;
            }
            set
            {
                _airbag = airbag;
            }
        }
        //Modificación del metodo precio
        public override int precio 
        { 
            get
            {
                if(airbag == true)
                {
                    return base.precio += 200;
                }
                else
                {
                    return base.precio;
                }
            }
        }
        
        //Modificando metodos padre
        public override string ToString()
        {
            if(airbag == true)
            {
                return base.ToString() + "\nSi tiene Airbags";
            }
            else
            {
                return base.ToString() + "\nNo tiene Airbags";
            }
        }
        
    }
}